File: php/src/DA/DAConsultaUsuarios.php
<?php
/** Acceso a datos para consultas de usuarios. Equivalente a DA/DAConsultaUsuarios.vb. */

require_once __DIR__ . '/../Db.php';

class DAConsultaUsuarios
{
    /** @return array<int,array<string,mixed>> */
    public function ConsultasUsuariosPrincipal(string $codigo, string $estado): array
    {
        return Db::callSp('webDatpos_consultaPorCodEmpresa', [$codigo, $estado]);
    }

    /** @return array<int,array<string,mixed>> */
    public function ConsultaUsuariosPorEmpresa(string $empresa): array
    {
        return Db::callSp('webDatpos_countUsuariosPorEmpresa', [$empresa]);
    }
}

File: DA/DAConsultaUsuarios.vb
Imports System.Data.SqlClient

Public Class DAConsultaUsuarios

    Dim objDAC As New DAConexionSQL()

    Public Function ConsultasUsuariosPrincipal(codigo As String, estado As String)

        Dim cmd As New SqlCommand
        cmd.Parameters.Add(New SqlParameter("@ccod_empresa", codigo))
        cmd.Parameters.Add(New SqlParameter("@cstatus", estado))
        cmd.CommandText = "webDatpos_consultaPorCodEmpresa"

        Return objDAC.selectstored(cmd)

    End Function
 

    Public Function ConsultaUsuariosPorEmpresa(empresa As String)

        Dim cmd As New SqlCommand
        cmd.Parameters.Add(New SqlParameter("@ccod_empresa", empresa))
        cmd.CommandText = "webDatpos_countUsuariosPorEmpresa"

        Return objDAC.selectstored(cmd)

    End Function


End Class
